#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_check_apt_return_unknown.dpatch by Jan Wagner <waja@cyconet.org>
##
## DP: Return UNKNOWN, if we don't know what happens (#498812)

@DPATCH@
--- nagios-plugins-1.4.14~/plugins/check_apt.c	2009-10-16 00:20:03.000000000 +0200
+++ nagios-plugins-1.4.14/plugins/check_apt.c	2009-10-16 00:20:38.000000000 +0200
@@ -291,7 +291,7 @@
 	/* If we get anything on stderr, at least set warning */
 	if(chld_err.buflen){
 		stderr_warning=1;
-		result = max_state(result, STATE_WARNING);
+		result = max_state_alt(result, STATE_UNKNOWN);
 		if(verbose){
 			for(i = 0; i < chld_err.lines; i++) {
 				fprintf(stderr, "%s\n", chld_err.line[i]);
